arch/arc/core/fault.c

```diff
@@ -90,6 +94,9 @@ static bool z_check_thread_stack_fail(const uint32_t fault_addr, uint32_t sp)
 		guard_end = thread->stack_info.start;
 		guard_start = guard_end - Z_ARC_STACK_GUARD_SIZE;
 	}
+#else /* CONFIG_MULTITHREADING */
+	guard_start = (uint32_t)z_main_stack - Z_ARC_STACK_GUARD_SIZE;
+	guard_end = (uint32_t)z_main_stack;
```

Comment:
Hmm, from the first look it should be something like this:

```suggestion
	guard_end = (uint32_t)z_main_stack + Z_ARC_STACK_GUARD_SIZE;
	guard_start = (uint32_t)z_main_stack;
```

Reply:
@evgeniy-paltsev thank you, I have no idea what the stack layout is or where the stack guards are on ARC. I will therefore defer to you, but I will note that the other stacks assume the stack guard to be right before the stack if I read the code correctly:

```
/* Special case: handling a syscall on privilege stack.
 * There is guard memory reserved immediately before
 * it.
 */
guard_end = thread->arch.priv_stack_start;
guard_start = guard_end - Z_ARC_STACK_GUARD_SIZE;
```
and
```
/* Supervisor thread */
guard_end = thread->stack_info.start;
guard_start = guard_end - Z_ARC_STACK_GUARD_SIZE;
```

Reply:
this would suggest:
```
guard_end = (uint32_t)z_main_stack;
guard_start = guard_end - Z_ARC_STACK_GUARD_SIZE;
```